packages/terra-worklist-data-grid/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -183,19 +187,28 @@ const ColumnHeaderCell = (props) => {
   // Retrieve current theme from context
   const theme = useContext(ThemeContext);
 
+  const pinnedColumnsDivider = columnIndex === columnContext.pinnedColumnsLength - 1
+    ? (
+      <div
+        className={cx('pinned-columns-divider')}
+        style={{ height: tableHeight, left: width - 1 }} // eslint-disable-line react/forbid-dom-props
+      />
+    )
+    : null;
+
   return (
   /* eslint-disable react/forbid-dom-props */
     <th
       ref={(columnHeaderCellRef)}
       key={id}
-      className={cx('column-header', theme.className, { selectable: isSelectable })}
+      className={cx('column-header', theme.className, { selectable: isSelectable, pinned: columnIndex < columnContext.pinnedColumnsLength })}
       tabIndex={isTabStop ? 0 : -1}
       role="columnheader"
       scope="col"
       aria-sort={sortIndicator}
       onMouseDown={(isSelectable && onColumnSelect) ? handleMouseDown : undefined}
       onKeyDown={(isSelectable && onColumnSelect) ? handleKeyDown : undefined}
-      style={{ width: `${width}px`, height: headerHeight }}
+      style={{ width: `${width}px`, height: headerHeight, left: columnContext.pinnedColumnOffsets[columnIndex] }} // eslint-disable-line react/forbid-dom-props
```

Comment:
I would think that the left style would not be added unless this is a pinned column.  Do we want to access invalid array indexes to provide this implementation?

Reply:
Updated: [9ac76fa](https://github.com/cerner/terra-framework/pull/1710/commits/9ac76faf9f340173a01fa8654ec5362ff045eaf1)